### ISSUE TITLE
Having to create a new workspace sometime

### ISSUE THREAD

**Issue by agrignard** [gama-platform/gama#1721]
Well I don't know if it's a real issue and it's quiet hard to document but I think it would be nice to fix that for a more user friendly UI. Maybe it happens only in the developing version and not in the realease. Anyway...

As a developer we are used to change very often our workspace, some of us do use name like Workspace2850, Workspace2851, etc etc

This is ok as soon as you are just developing new features in GAMA, but when you spend several days on your workspace with all your personal preference and model you really don't want to change your workspace.

So it happens to me today, for an unknown reason I cannot launch GAMA with the current workspace.
The only thing I can provide is this splashcreen where GAMA remains blocked on this stage, and I really don't know what I did wrong except developping my own model. 

So the only solution is to create a new workspace and so redefine my personal preference, find a new name like MyWorkspace#2 and reimport your project, etc... I know it's not a big deal but it would be nice not to have to do it.
![screen shot 2016-05-03 at 11 30 19](https://cloud.githubusercontent.com/assets/3928502/14988865/c7357ba2-1123-11e6-938c-6283d298cc78.png)

**Comment by AlexisDrogoul** [comment]
Yep. Happens also in Eclipse from time to time. You can try adding to the application options `-clearPersistedState` when it happens, and then relaunch GAMA from the product. In case it is still stuck, you can manually delete the file `workbench.xmi` located at `.metadata/.plugins/org.eclipse.e4.workbench` in your workspace. 

This is the reason why I chose to store the preferences of GAMA (not the ones shared with the Eclipse foundation) globally -- they should be unaffected by a change of workspace. One thing we could do to ease the transition is

1. to always export the preferences of Eclipse in a given place in the workspace
2. to give the option, when creating a new workspace to import projects and the stored preferences from an existing workspace.

It would probably ease the pain caused by this issue. What do you think ?

**Comment by agrignard** [comment]
Ok. Not a big deal for me but for a new user I think it can be really frustrated to loose your workspace like that. So yes the 2 options you mention can be a good way to ease the pain.

Just to try to understand everything what do you mean by "application options", where should I add the -clearPersistedState? In the product/Launching?

I just deleted the workbench.xmi but I still cannot launch my old workspace (not a big deal but I just want to figure out the problem).

And if adding -clearPersistedState and workbench.xmi works, why not doing it automatically each time we run GAMA (well I guess workbench.xmi is here for a certain reason...)?

**Comment by AlexisDrogoul** [comment]
The persisted state and the `workbench.xmi` file are here to remember the current state of GAMA each time you launch it (open editors, etc.). 

`-clearPersistedState` should be put in Program Arguments in the "Launching" tab of the product. 

This box should look like:

```
-nl ${target.nl} 
-data @noDefault 
-clearPersistedState
--launcher.defaultAction
openFile
```

**Comment by agrignard** [comment]
Ok it works with -clearPersistedState but I get the point, we cannot do it automaticcaly as it clear everything.

**Comment by agrignard** [comment]
Sorry to insist on that but I'd like to be able to keep my workspace and I find it so frustrating to have eclipse doing something  in your back and not being able to understand how to fix it...

So I reget the same problem (like in the screenshot) even with -clearPersistedState and when I go to .metadata/.plugins/org.eclipse.e4.workbench there is nothing inside...

**Comment by AlexisDrogoul** [comment]
Well. It happened to me 1 or 2 times in the last months, and, believe me, I'm messing a lot with my workspace, so it's kinda normal ! You dont have other processes interfering with the files there ? Dropbox, GDrive, git, svn ? I dont know what causes these workspace corruptions -- and you dont have anything in your logs (not the console ones, but the "Errors" within eclipse) ? Could be interesting to enable tracing of the core eclipse plugins (ide, ui, etc.) to see what happens in these cases.

**Comment by AlexisDrogoul** [comment]
Among the things that can corrupt a workspace, two are the most cited:

- force-quitting the application (and its equivalent in Eclipse, clicking on the red square). It is always better (when possible, of course) to quit Gama normally.
- running two (or more) instances of Gama that share the same workspace. Not advised at all. 

With these two things in mind, It happens to me very rarely now (it seems to have even improved with Eclipse Mars).

**Comment by AlexisDrogoul** [comment]
I've committed a change in the handling of preferences that:

- save local preferences when exiting GAMA
- allows, when creating a new workspace, to clone an existing one (and not only the "current" one), including re-applying the preferences and copying the models over.

This means that, if a workspace is "stuck", you have to rename it on your hard disk and then launch GAMA. When invited to create a new one, keep the old name, but choose 'Close an existing workspace' and choose the one just renamed. All preferences and models should be copied over. 

There is nothing more I can do on this issue to support the user when this behavior happens, so I close it for now.
